### DISCUSSION TITLE
Reward

### DISCUSSION THREAD

**Discussion by Yiquan-lol** [praveen-palanisamy/macad-gym#42] [Q&A]
Hi,
Where is the reward function defined?
Is it in the [https://github.com/praveen-palanisamy/macad-gym/blob/master/src/macad_gym/carla/reward.py](url) file？

**Comment by praveen-palanisamy** [comment]
Hi @Yiquan-lol ,

The `Reward` class in [reward.py](https://github.com/praveen-palanisamy/macad-gym/blob/master/src/macad_gym/carla/reward.py) has methods for the different types (currently 4) of reward functions that can be configured to be used in a MACAD-Gym environment.

For example, if you are using the following environment (as in the README example):
```python
import gym
import macad_gym
env = gym.make("HomoNcomIndePOIntrxMASS3CTWN3-v0")
# Your agent code here
```

the `HomoNcomIndePOIntrxMASS3CTWN3-v0` environment's config uses the following reward function:
https://github.com/praveen-palanisamy/macad-gym/blob/38884ac4bc7fb2e91a865950cff4eeadeed02f60/src/macad_gym/envs/homo/ncom/inde/po/intrx/ma/stop_sign_3c_town03.py#L56
which is defined here:
https://github.com/praveen-palanisamy/macad-gym/blob/38884ac4bc7fb2e91a865950cff4eeadeed02f60/src/macad_gym/carla/reward.py#L43-L67

**Comment by praveen-palanisamy** [comment]
Converting this to a discussion.
